prov/shm/src/smr_ep.c

```diff
@@ -216,6 +216,10 @@ static int smr_match_unexp(struct dlist_entry *item, const void *args)
 	struct smr_unexp_msg *unexp_msg;
 
 	unexp_msg = container_of(item, struct smr_unexp_msg, entry);
+	if (unexp_msg->cmd.msg.hdr.op == ofi_op_msg)
+		return smr_match_addr(unexp_msg->cmd.msg.hdr.addr, attr->addr);
```

Comment:
How do we know that we won't match an untagged message with a tagged message?  Here the incoming message is untagged, but how do we know that the entry we're matching against isn't tagged?

Reply:
You are right, I will submit a new revision.